### PR TITLE
internal: Shrink size of hir::Binding

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

File: crates/hir-def/src/body.rs [hunk 1]
```diff
@@ -257,6 +270,17 @@ impl Body {
...
     }
+
+    pub fn is_binding_upvar(&self, binding: BindingId, relative_to: ExprId) -> bool {
+        match self.binding_owners.get(&binding) {
+            Some(x) => {
+                // We assign expression ids in a way that outer closures will receive
+                // a lower id
+                x.into_raw() < relative_to.into_raw()
+            }
+            None => true,
```
Comment (Veykril) [inline crates/hir-def/src/body.rs:281]: @HKalbasi this is unchanged, but shouldn't the `None` branch here return `false`? Or am I misunderstanding this function

Comment (HKalbasi) [reply crates/hir-def/src/body.rs:281]: In case of `None` the binding is defined in the root of body, so it is upvar to any closure.

Comment (Veykril) [reply crates/hir-def/src/body.rs:281]: Ah, okay I was misunderstanding then, thanks

**Pull request merged**